src/components/VariantSearchField.tsx

```diff
@@ -109,6 +121,49 @@ export const VariantSearchField = ({ onVariantSelect, currentSelection, triggerS
     }
   }, [currentSelection]);
 
+  useEffect(() => {
+    if (selectedOptions1 != undefined && selectedOptions1.length > 0) {
+      setAdvancedSearch(false);
+      setVariantQuery('');
+      setSelectedOptions(selectedOptions1);
+    } else if (variantQuery1 !== undefined && variantQuery1.length > 0) {
+      setAdvancedSearch(true);
+      setSelectedOptions([]);
+      setVariantQuery(variantQuery1);
+    }
+  }, [variantQuery1, selectedOptions1]);
+
+  // useEffect(() => {
```

Comment:
Please avoid committing commented-out code unless there is a good (and documented!) reason.